programs/virtual-curve/src/instructions/migration/create_locker.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{
+    constants::seeds::{BASE_LOCKER_PREFIX, POOL_AUTHORITY_PREFIX},
+    state::{MigrationProgress, PoolConfig, VirtualPool},
+    *,
+};
+use anchor_lang::solana_program::{program::invoke, system_instruction};
+use anchor_spl::{
+    token_2022::{mint_to, MintTo},
+    token_interface::{TokenAccount, TokenInterface},
+};
+use locker::cpi::accounts::CreateVestingEscrowV2;
+
+#[derive(Accounts)]
+pub struct CreateLockerCtx<'info> {
+    /// Config
+    #[account(mut, has_one = config, has_one = creator)]
+    pub virtual_pool: AccountLoader<'info, VirtualPool>,
+    /// Config
+    pub config: AccountLoader<'info, PoolConfig>,
+    /// CHECK: pool authority
+    #[account(
+        mut,
+            seeds = [
+                POOL_AUTHORITY_PREFIX.as_ref(),
+            ],
+            bump,
+        )]
+    pub pool_authority: UncheckedAccount<'info>,
+    /// CHECK: base_vault
+    #[account(
+        mut,
+        token::mint = base_mint,
+        token::token_program = token_program
+    )]
+    pub base_vault: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// CHECK: base token mint
+    #[account(mut)]
+    pub base_mint: UncheckedAccount<'info>,
+
+    /// CHECK: base key to create locked escrow
+    #[account(
+        mut,
+        seeds = [
+            BASE_LOCKER_PREFIX.as_ref(),
+            virtual_pool.key().as_ref(),
+        ],
+        bump,
+    )]
+    pub base: UncheckedAccount<'info>,
+    /// CHECK: owner
+    pub creator: UncheckedAccount<'info>,
+    /// CHECK: escrow of locker, derived from base
+    #[account(mut)]
+    pub escrow: UncheckedAccount<'info>,
+
+    /// CHECK: escrow token
+    #[account(mut)]
+    pub escrow_token: UncheckedAccount<'info>,
+
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// CHECK: token_program
+    pub token_program: Interface<'info, TokenInterface>,
+
+    /// CHECK: Locker program
+    #[account(address = locker::ID)]
+    pub locker_program: UncheckedAccount<'info>,
+
+    /// CHECK: Locker event authority
+    pub locker_event_authority: UncheckedAccount<'info>,
+
+    /// System program.
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handle_create_locker(ctx: Context<CreateLockerCtx>) -> Result<()> {
+    let mut virtual_pool = ctx.accounts.virtual_pool.load_mut()?;
+
+    require!(
+        virtual_pool.get_migration_progress()? == MigrationProgress::PostBondingCurve,
+        PoolError::NotPermitToDoThisAction
+    );
+
+    let config = ctx.accounts.config.load()?;
+
+    let locked_vesting_params = config.locked_vesting_config.to_locked_vesting_params();
+    let total_vested_amount = locked_vesting_params.get_total_amount()?;
+
+    let pool_authority_seeds = pool_authority_seeds!(ctx.bumps.pool_authority);
+
+    msg!("mint token to base vault");
+    mint_to(
```

Comment:
Why don't mint the vesting amount during initialization of virtual curve?

Reply:
that is fine, because jup-lock is only created if token is migrated
- Less token supply in pre-bonding-curve
- Dont affect on other logics

Reply:
Personally, think that it's better for it to be minted during initialization of the virtual curve. That make user / integrator easier since the `mint.supply` will be the token supply even before migration to the AMM. But, I guess it's fine.